### ISSUE TITLE
Datadog chart, replicaset not updated appropriately

### ISSUE THREAD

**Issue by janbrunrasmussen** [helm/charts#10005]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
HELM:
Client: &version.Version{SemVer:"v2.12.0", GitCommit:"d325d2a9c179b33af1a024cdb5a4472b6288016a", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.0", GitCommit:"d325d2a9c179b33af1a024cdb5a4472b6288016a", GitTreeState:"clean"}

Kubernetes:
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-04T07:51:55Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.5", GitCommit:"753b2dbc622f5cc417845f0ff8a77f539a4213ea", GitTreeState:"clean", BuildDate:"2018-11-26T14:31:35Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

**Which chart**:
stable/datadog

**What happened**:
The problem in this issue: https://github.com/helm/helm/issues/1844
Basically changes to e.g. image version or other values in the cluster-agent deployment, creates a new replicaset, without scaling the old one to 0. This means that values are not always passed on to the new deployment.

**What you expected to happen**:
"old" replicaset is scaled to zero and values are updated.

**How to reproduce it** (as minimally and precisely as possible):
Install chart
```
$ helm upgrade datadog stable/datadog \
                --namespace datadog \
                --set clusterAgent.enabled=true \
                --install \
                --recreate-pods
```
Edit release
```
$ helm upgrade datadog stable/datadog \
                --namespace datadog \
                --set clusterAgent.enabled=true --set clusterAgent.image.tag=1.0.1 \
                --install \
                --recreate-pods
```
Get replicasets in namespace
```
$ kubectl get replicaset -n datadog
NAME                                      DESIRED   CURRENT   READY   AGE
datadog-cluster-agent-5cb665c9b5        1         1         0       8s
datadog-cluster-agent-7d65769f86        1         1         1       2m
datadog-kube-state-metrics-58d5695f4f   1         1         0       2m
```
Note `DESIRED` is `1` for both replicasets.

**Anything else we need to know**:
I guess this is where the problem lie, but I have _not_ verified it: https://github.com/helm/charts/blob/master/stable/datadog/templates/cluster-agent-deployment.yaml#L13-L19

**Comment by xvello** [comment]
Hi @janbrunrasmussen ,

I can reproduce it with your command, but the issue seems that `datadog/cluster-agent:1.0.1` has not been published. If I use `--set clusterAgent.image.tag=1.0.0` the update goes OK. Could you please try?

1.0.1 has only been released as `datadog/cluster-agent-dev:1.0.1`, I will investigate why.

**Comment by janbrunrasmussen** [comment]
Thanks for looking at this! Try any other change, and there is a fair chance this behaviour will still be there. We originally saw it when trying to change the log level, so I still think it is related to the chart, not the image version.
